cosmo_tester/resources/blueprints/component/component.yaml

```diff
@@ -0,0 +1,70 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+description: >
+  Create a file in an arbitrary location on a pre-existing VM using a plugin.
+imports:
+  - http://www.getcloudify.org/spec/cloudify/5.1.0.dev1/types.yaml
+
+inputs:
+  server_ip:
+    default: 127.0.0.1
+  agent_user:
+    default: centos
+  path: {}
+  content: {}
+  os_family:
+    default: linux
+  agent_port:
+    default: 22
+  agent_password:
+    default: ""
+
+node_templates:
+  infrastructure:
+    type: cloudify.nodes.Component
+    properties:
+      resource_config:
+        blueprint:
+          id: infra
+          main_file_name: 'fake-infra.yaml'
+          external_resource: true
+        deployment:
+          inputs:
+            agent_user: { get_input: agent_user }
+            path: { get_input: path }
+
+  vm:
+  # Represents an application.
+    type: cloudify.nodes.Compute
+    properties:
+      ip: { get_input: server_ip }
```

Comment:
Shouldn't these properties be populated by capabilities of the component? (see capabilities test and blueprints- they might cover more than just the one I spotted below)

Reply:
I don't test capabilities in this test, and it will make re-using _app, _infra creators for shared resource impossible